tests/components/tod/test_binary_sensor.py

```diff
@@ -652,7 +653,7 @@ async def test_norwegian_case_summer(hass):
         get_astral_event_next(hass, "sunrise", dt_util.as_utc(test_time))
     )
     sunset = dt_util.as_local(
-        get_astral_event_next(hass, "sunset", dt_util.as_utc(test_time))
+        get_astral_event_next(hass, "sunset", dt_util.as_utc(sunrise))
```

Comment:
Why do we pass sunrise instead of test_time?

Reply:
In norway in june on the day 2010-06-01 the next event is a sunset, so the sunset event would be happening before the sunrise. As such for what the rest is written to test, we want the sunset that happens after the next sunrise. As the tests are intended to check.

This somehow passed through previously as the previous astral just missed by a few hours the sunrise that actually happens after the test case.

In practice our astral should always be getting the next sunset and sunrise when calling "get_astral_event_next" and as modules update on 'last event' in normal practice ( not the test ) the module would actually be re-evaluating the sunset time at the sunrise anyway.